bottlecap/src/secrets/decrypt.rs

```diff
@@ -53,7 +53,18 @@ pub async fn resolve_secrets(config: Arc<Config>, aws_config: &mut AwsConfig) ->
             }
 
             let decrypted_key = if config.kms_api_key.is_empty() {
-                decrypt_aws_sm(&client, config.api_key_secret_arn.clone(), aws_config).await
+                let secret_region = config
+                    .api_key_secret_arn
+                    .split(':')
+                    .nth(3)
+                    .unwrap_or(&aws_config.region);
+                decrypt_aws_sm(
+                    &client,
+                    config.api_key_secret_arn.clone(),
+                    aws_config,
+                    secret_region.to_string(),
+                )
+                .await
             } else {
                 decrypt_aws_kms(&client, config.kms_api_key.clone(), aws_config).await
```

Comment:
Would KMS have the same issue? we are sending the `aws_config` which has the region config for the lambda